torch/fx/graph.py

```diff
@@ -1010,16 +1010,28 @@ def check_arg(arg : Node, n : Optional[Node] = None) -> None:
                     target_atoms = node.target.split('.')
                     m_itr = self.owning_module
                     for i, atom in enumerate(target_atoms):
-                        m_itr = getattr(m_itr, atom, None)
-                        if m_itr is None:
-                            seen_qualname = '.'.join(target_atoms[:i])
+                        new_m_itr = getattr(m_itr, atom, None)
+                        seen_qualname = '.'.join(target_atoms[:i])
+                        if new_m_itr is None:
                             raise RuntimeError(f'Node {node} target {node.target} references nonexistent attribute '
                                                f'{atom} of {seen_qualname}')
+                        if (node.op == "call_module"
+                                and not isinstance(new_m_itr, torch.nn.Module)):
+                            raise RuntimeError(f'Node {node} target {node.target} {atom} of {seen_qualname} does '
+                                               'not reference an nn.Module')
+                        elif (node.op == "get_attr"
+                                    and not isinstance(new_m_itr, torch.nn.Module)
+                                    and not isinstance(new_m_itr, torch.nn.Parameter)
+                                    and atom not in m_itr._buffers):
+                                raise RuntimeError(f'Node {node} target {node.target} {atom} of {seen_qualname} does '
+                                                    'not reference an nn.Module')
```

Comment:
Hard to tell because of lint failures, but I think the error message is incorrect. The check checks if the value is a module, parameter, or buffer, but the message says `nn.Module`.


BTW, I don't know if this should apply to `get_attr`. It's valid to have a `get_attr` to a non-module/parameter/buffer, e.g. see the native interpreter example https://github.com/pytorch/examples/blob/b0649dcd638eb553238cdd994127fd40c8d9a93a/fx/native_interpreter/use_interpreter.py#L100